### PR TITLE
fix missing doc Generic Reader

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description

Add import with functors in GenericReader in main default reader  (mainly motivated to show documentation of specialized version of  importWithValueFunctor and importWithColorFunctor).

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by kerautret** [comment]
I also fix some extension to read other ITK based format in link to DGtalTools PR #296

File: ChangeLog.md [hunk 1]
```diff
@@ -17,6 +17,13 @@
  - Fix testBasicPointFunctor. (Bertrand Kerautret
    [#1245](https://github.com/DGtal-team/DGtal/pull/1245))
 
+- *Documentation*
+ - Add import with functors in GenericReader in main default reader.
```
Comment (dcoeurjo) [inline ChangeLog.md]: in the main

File: src/DGtal/io/readers/GenericReader.h [hunk 2]
```diff
@@ -112,6 +112,78 @@ namespace DGtal
...
     static TContainer import(const std::string &filename,
                              std::vector<unsigned int> dimSpace= std::vector<unsigned int > () )  throw(DGtal::IOException);
+
+    
+    /**
+     * Import  a volume nd image file by specifying a value functor.
+     *
+     * @tparam TFunctor The type of the functor (should verify the concept CUnaryFunctor<TFunctor, TValue , TContainer::Value > )     * @param dimSpace a vector containing the n dimensional image size..
+     * @param filename the image filename to be imported.
+     * @param aFunctor to transform input type of image value (TValue) into the given image type (TContainer::Value).
```
Comment (dcoeurjo) [inline src/DGtal/io/readers/GenericReader.h]: to transform THE ..

Comment (dcoeurjo) [inline src/DGtal/io/readers/GenericReader.h]: please 80col for the doxygen doc

File: src/DGtal/io/readers/GenericReader.h [hunk 3]
```diff
@@ -112,6 +112,78 @@ namespace DGtal
...
+        
+        trace.error() << "Extension " << extension<< " not yet implemented in DGtal GenericReader." << std::endl;
+        throw dgtalio;
+      }
+
+    
+    /**
+     * Import  a volume nd image file by specifying a value functor.
+     *
+     * @tparam TFunctor The type of the functor (should verify the concept CUnaryFunctor<TFunctor, TValue , TContainer::Value > )     * @param dimSpace a vector containing the n dimensional image size..
```
Comment (dcoeurjo) [inline src/DGtal/io/readers/GenericReader.h]: 80col

**Comment by kerautret** [comment]
done thanks @dcoeurjo

**Comment by dcoeurjo** [comment]
thanks merging

**Pull request merged**